elements/lisk-api-client/src/ws_channel.ts

```diff
@@ -108,15 +111,33 @@ export class WSChannel {
 			return Promise.resolve();
 		}
 
-		return new Promise<void>(resolve => {
-			this._ws?.on('close', () => {
-				this._ws?.terminate();
-				this.isAlive = false;
-				this._ws = undefined;
-				resolve();
-			});
-			this._ws?.close();
+		if (this._ws.readyState === WebSocket.CLOSED) {
+			this.isAlive = false;
+			this._ws = undefined;
+			return Promise.resolve();
+		}
+
+		const disconnect = new Promise<void>(resolve => {
+			const retry = () => {
+				const id = setTimeout(() => {
+					clearTimeout(id);
+					if (!this.isAlive) {
+						return resolve();
+					}
+					return retry();
+				}, 100);
+			};
+
+			retry();
 		});
+
+		this._ws.close();
+		await Promise.race([
+			disconnect,
+			timeout(CONNECTION_TIMEOUT, `Could not disconnect in ${CONNECTION_TIMEOUT}ms`),
+		]);
+
+		return Promise.resolve();
```

Comment:
same as above

Reply:
Same here its some ESLint issue that it complaints about consistent promise return.